lib/public/UrlChecker.js

```diff
@@ -63,24 +63,40 @@ UrlChecker.prototype.dequeue = function(id)
 
 UrlChecker.prototype.enqueue = function(url, baseUrl, customData)
 {
+	var testUniqueUrl = (function(url) {
+		if (this.options.checkUniqueUrlOnce===true) {
+			if (this.urlCache.indexOf(url) > -1) {
+				return false;
+			} else {
+				this.urlCache.push(url);
+				return true;
+			}
+		}
+		return true;
+	}).bind(this);
```

Comment:
There should also be a `clearCache()` method in this class and pointed to from `HtmlUrlChecker` and `HtmlChecker`